### PR TITLE
Messaging tweaked and bug in roof push fixed

### LINKED ISSUE #1283: Warnings and errors missing
#### Description:

As in the title, a few missing messages have been spotted while testing the toolkit:
- no error in case of push of a disjoint wall/floor/roof
- no warning about the wall/roof openings being skipped

On top of that, sometimes the family library temp folder gets stuck and needs cleanup - would be good to capture it in the message too.

### LINKED ISSUE #1284: Pushing of roofs with holes ignores holes
#### Description:

If the location of a BHoM roof is defined with an inner outline (hole), the latter gets ignored.

### LINKED ISSUE #1286: Null exception in Revit 2023 when pulling roofs
#### Description:

Using the test files below for pulling roofs from Revit 2023, I got this error:

image

which probably comes from here:

image

#### Steps to reproduce:

Simply open both test files, select roof ID 1297978 and perform a pull. The same error probably occurs on all other roofs.

#### Expected behaviour:

All roofs should pull without error.

#### Test file(s):

On [SharePoint](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/Revit_Toolkit/Revit_Core_Engine-%231286-NulExceptionOnPullingRoofs?csf=1&web=1&e=t6J5ac)

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #1283
Closes #1284
Closes #1286

### Test files

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FRevit%5FToolkit%2F%231285%2DMessageTweaks&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4)

### CODE REVIEW

**Comment by pawelbaran** [comment]
Oke, I have pushed one more commit + added new test file. Ready for review now 👍

**Comment by vietle-bh** [comment]
> Oke, I have pushed one more commit + added new test file. Ready for review now 👍

Could you also list which GH scripts we should use to test this?

**Review by vietle-bh — COMMENTED** [review]
It's a huge improvement! This is before the fix:
image
image

And this is after:
image
image

The remaining errors seem to be around here;
image

Will we fix this in this PR?

File: Revit_Core_Engine/Convert/Physical/ToRevit/Wall.cs [hunk 1]
```diff
@@ -82,6 +86,12 @@ public static Wall ToRevitWall(this oM.Physical.Elements.Wall wall, Document doc
             if (revitWall == null)
                 return null;
 
+            document.Regenerate();
```
[the remark this answers is not in the document]
Comment (pawelbaran) [reply Revit_Core_Engine/Convert/Physical/ToRevit/Wall.cs]: No, it is redundant actually - legacy after another attempt. Well spotted!

**Comment by pawelbaran** [comment]
Thanks @vietle-bh for review. I have addressed all of your comments, should be ready for re-review now

**Comment by vietle-bh** [comment]
> Thanks @vietle-bh for review. I have addressed all of your comments, should be ready for re-review now

Great! Which Grasshopper script should I use to test the Push though?
(I tested before only with a Pull script)

**Comment by vietle-bh** [comment]
By the way @pawelbaran , I found a related issue while testing: #1286 
It would be good to fix it too 😅

**Comment by pawelbaran** [comment]
> Great! Which Grasshopper script should I use to test the Push though? (I tested before only with a Pull script)

_#1285-MessageTweaks_  shows the tweaks to push, _#1285-PullPanelFix_  shows the tweaks to pull

**Comment by pawelbaran** [comment]
I have also fixed #1286 @vietle-bh - I hope all covered now 🙈

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check core

**Comment by pawelbaran** [comment]
@BHoMBot check required

**Comment by vietle-bh** [comment]
> I have also fixed #1286 @vietle-bh - I hope all covered now 🙈

There're definitely fewer errors now 👍 However, some items fail on another null exception due to nurbs curves in their boundaries.
image

image

Error 1 already gives enough info so it would be good to maybe hide the second error:

image

**Review by vietle-bh — APPROVED** [review]
Approved based on the conversation above.

**Comment by pawelbaran** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check ready-to-merge

**Pull request merged**